pandas/core/reshape/pivot.py

```diff
@@ -79,6 +79,13 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
                 pass
         values = list(values)
 
+    num_rows = (data.reindex(columns=index).drop_duplicates().shape[0]
+                if index else 1)
+    num_cols = (data.reindex(columns=columns).drop_duplicates().shape[0]
+                if columns else 1)
```

Comment:
The goal here is to get an accurate size of the resulting pivot table, taking into account potential duplicates.

However I recognize that the performance hit may not be worth it given that this is such an edge case. It's up to you whether we should just raise the error within `unstack`.